resources/service.rb

```diff
@@ -43,11 +43,15 @@
 action_class do
   def do_service_action(resource_action)
     with_run_context(:root) do
-      edit_resource(:execute, "Run pre service #{resource_action} #{dhcpd_service_name(new_resource.ip_version)} configuration test.") do
-        command dhcpd_config_test_command(new_resource.ip_version, new_resource.config_file)
-        only_if { new_resource.config_test && %i(start restart reload).include?(resource_action) && ::File.exist?(new_resource.config_file) }
-        user dhcpd_user
-        action :nothing
+      begin
+        edit_resource(:execute, "Run pre service #{resource_action} #{dhcpd_service_name(new_resource.ip_version)} configuration test.") do
+          command dhcpd_config_test_command(new_resource.ip_version, new_resource.config_file)
+          only_if { new_resource.config_test && %i(start restart reload).include?(resource_action) && ::File.exist?(new_resource.config_file) }
+          user dhcpd_user
+          action :nothing
+        end
+      rescue Mixlib::ShellOut::ShellCommandFailed
+        delete_resource!(:service, new_resource.service_name.delete_suffix('.service'))
```

Comment:
why not just do a Chef::Log.error "Someone gone messed up their config"

Reply:
Because I don't want it to restart the service and leave a borked/failed dhcp service. The implicit raise from the configuration file test failure doesn't always stop the delayed service restart.

Reply:
What about at least doing a Chef::Log.warn so the user can hopefully see why that happened?

Reply:
They're going to see the stack trace from the failed `ShellCommandFailed` with the stderr output from dhcpd with the reason why it failed a configuration test so I'm not sure that would be necessary or even easily visible.